lib/config/lib/config.c

```diff
@@ -40,18 +40,18 @@ config_open(const char *const path, config_validator_t validate, cJSON **const c
         goto out;
     }
 
-    str = malloc(st.st_size);
+    str = malloc((size_t)st.st_size);
```

Comment:
Signed to unsigned.

Reply:
So, it's broken with or without the cast, whereas the cast only serves to silence the warning.  My issue is that going down this path (enable warnings then shut them all off with casts) is going to lead to a ton of casts all over the place.  So instead of casts alerting the read to pay attention, there are so many that none are special...

Reply:
In this particular instance, would you like it more if I saved it off into a temporary variable to change 3 casts into 1?

I definitely agree with you generally. I just feel like the codebase has a lot of places where we change sign without purpose that are just flying under the radar, so you can't catch those other locations without having to deal with these stupid situations. It is unfortunate that we have to live with the baggage of POSIX, UNIX, and Linux like this signed `st_size`.